Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2697 -- Select category/subcategory more easily

Resolves [EREGCSC-2697](https://jiraent.cms.gov/browse/EREGCSC-2697)

**Description**

Categories and subcategories are in order in a way that makes them easier to select

**This pull request changes:**

- Refactors category models so that `name` and `parent` fields are added to `AbstractCategory`
- Adds leading `&nbsp;` spaces to subcategory options to visually communicate subcategory relationship
- Sorts categories and subcategories alphabetically in the select component

**Steps to manually verify this change:**

1. Green check marks
2. Visit Admin panel on the [experimental deployment](https://gcuzpozzwl.execute-api.us-east-1.amazonaws.com/dev1390)
3. Add a file or link and ensure that the category picker looks as expected
